server/services/health_service.py

```diff
@@ -85,101 +87,149 @@ def _load_seed_runtime_model() -> tuple[str | None, str]:
-    try:
-        resp = requests.get(url, timeout=1.0)
-        if 200 <= resp.status_code < 300:
-            return {
-                "name": "ollama",
-                "status": "healthy",
-                "message": f"{base} reachable",
-                "latency_ms": _now_ms() - start,
-                "details": {"host": base},
-            }
-        return {
-            "name": "ollama",
-            "status": "unhealthy",
-            "message": f"{base} returned {resp.status_code}",
-            "latency_ms": _now_ms() - start,
-            "details": {"host": base},
-        }
-    except Exception as e:
-        return {
-            "name": "ollama",
-            "status": "unhealthy",
-            "message": f"{base} unreachable: {e}",
-            "latency_ms": _now_ms() - start,
-            "details": {"host": base},
-        }
+    """Check Ollama runtime health using provider registry."""
+    provider = get_provider(Provider.ollama)
+    return provider.check_health({"base_url": settings.ollama_host})
+
+
+def _check_lemonade(port: int = 11534) -> dict:
+    """Check Lemonade runtime health using provider registry.
+
+    Args:
+        port: Lemonade server port (default: 11534)
+    """
+    provider = get_provider(Provider.lemonade)
+    return provider.check_health({"port": port})
 
 
 def _check_seed_project() -> dict:
-    """Validate the project seed runtime: Ollama is reachable and model is present."""
+    """Validate the project seed runtime is reachable and model is present using provider registry."""
     start = _now_ms()
-    model, reason = _load_seed_runtime_model()
-    base = settings.ollama_host.rstrip("/")
-    url = f"{base}/api/tags"
+    provider_str, model, port, reason = _load_seed_runtime_config()
 
     if model is None:
         return {
             "name": "project",
             "status": "unhealthy",
             "message": reason,
             "latency_ms": _now_ms() - start,
-            "runtime": {"host": base, "model": None},
+            "runtime": {"provider": provider_str, "model": None},
         }
 
-    # check tags for model
+    # Use provider registry for health checks
     try:
-        resp = requests.get(url, timeout=1.5)
-        ok = 200 <= resp.status_code < 300
-        if not ok:
+        # Map provider string to enum
+        provider_enum = Provider(provider_str)
+        provider_impl = get_provider(provider_enum)
+
+        # Build provider-specific config
+        config = {}
+        if provider_str == "ollama":
+            config["base_url"] = settings.ollama_host
+        elif provider_str == "lemonade":
+            config["port"] = port if port else 11534
+
+        # Get health check from provider
+        health_result = provider_impl.check_health(config)
+
+        # Enhance with model validation for Ollama
+        if provider_str == "ollama" and health_result.get("status") == "healthy":
+            details = health_result.get("details", {})
```

Comment:
If you use a well-known class as the response for check_health() you will not need to do all of this dictionary `.get()` junk